### PR TITLE
Account for NULL error when we see G_IO_STATUS_EOF

### PR DESCRIPTION
**Pull request by pyhalov**

### CODE REVIEW

**Comment by pyhalov** [comment]
On EOF error is NULL.

**Comment by lukefromdc** [comment]
What does this fix? I have no idea how to test it?

**Comment by pyhalov** [comment]
I couldn't reproduce this error without #383  , but with #383 it happens depending on pam configuration or when https://github.com/mate-desktop/mate-control-center/pull/383/commits/6d666384682dd0c4a4ec5e68f79164c8b9298e61 is applied on successful password change (when child exits).

**Comment by lukefromdc** [comment]
I don't understand this well, asking others. @monsta ? @raveit65 ?

**Comment by pyhalov** [comment]
As I see from https://github.com/GNOME/glib/blob/master/glib/giochannel.c#L1771 , on EOF error can be not set.

**Comment by pyhalov** [comment]
The issue looks like
```
leoric@openindiana:~/srcs/c_srcs$ sudo mdb  /var/cores/mate-about-me.2578 
Loading modules: [ libc.so.1 libnvpair.so.1 libuutil.so.1 libavl.so.1 ld.so.1 ]
> $C
fffffd7fffdfdd50 io_watch_stdout+0x5a()
fffffd7fffdfddd0 libglib-2.0.so.0.5701.0`g_main_context_dispatch+0x138()
fffffd7fffdfde30 libglib-2.0.so.0.5701.0`g_main_context_iterate.isra.26+0x1d8()
fffffd7fffdfde60 libglib-2.0.so.0.5701.0`g_main_loop_run+0xd2()
fffffd7fffdfded0 libgtk-3.so.0.2400.1`gtk_dialog_run+0x163()
fffffd7fffdfdf30 mate_about_me_password+0x3ae()
fffffd7fffdfdfc0 libgobject-2.0.so.0.5701.0`_g_closure_invoke_va+0x155()
fffffd7fffdfe110 libgobject-2.0.so.0.5701.0`g_signal_emit_valist+0xbcc()
fffffd7fffdfe1f0 libgobject-2.0.so.0.5701.0`g_signal_emit+0x7d()
fffffd7fffdfe210 libgtk-3.so.0.2400.1`gtk_button_do_release+0x4d()
fffffd7fffdfe260 libgtk-3.so.0.2400.1`gtk_real_button_released+0x3c()
fffffd7fffdfe2f0 libgobject-2.0.so.0.5701.0`_g_closure_invoke_va+0x155()
fffffd7fffdfe440 libgobject-2.0.so.0.5701.0`g_signal_emit_valist+0xbcc()
fffffd7fffdfe520 libgobject-2.0.so.0.5701.0`g_signal_emit+0x7d()
fffffd7fffdfe550 libgtk-3.so.0.2400.1`multipress_released_cb+0x29()
fffffd7fffdfe560 libffi.so.6.0.4`ffi_call_unix64+0x4c()
0000000000000800 0xfffffd7fffdfe710()
> io_watch_stdout+0x5a::dis
io_watch_stdout+0x31:           leaq   -0x60(%rbp),%rsi
io_watch_stdout+0x35:           movl   $0x40,%edx
io_watch_stdout+0x3a:           movq   %rbx,%rdi
io_watch_stdout+0x3d:           call   -0x1c82  <PLT=libglib-2.0.so.0.5701.0`g_io_channel_read_chars>
io_watch_stdout+0x42:           cmpl   $0x1,%eax
io_watch_stdout+0x45:           je     +0x39    <io_watch_stdout+0x80>
io_watch_stdout+0x47:           movq   -0x78(%rbp),%rax
io_watch_stdout+0x4b:           movl   $0x40ea4e,%edi
io_watch_stdout+0x50:           movl   $0x40ec07,%edx
io_watch_stdout+0x55:           movl   $0x10,%esi
io_watch_stdout+0x5a:           movq   0x8(%rax),%rcx
io_watch_stdout+0x5e:           xorl   %eax,%eax
io_watch_stdout+0x60:           call   -0x2055  <PLT=libglib-2.0.so.0.5701.0`g_log>
io_watch_stdout+0x65:           movq   -0x78(%rbp),%rdi
io_watch_stdout+0x69:           call   -0x204e  <PLT=libglib-2.0.so.0.5701.0`g_error_free>
io_watch_stdout+0x6e:           leaq   -0x18(%rbp),%rsp
io_watch_stdout+0x72:           movl   $0x1,%eax
io_watch_stdout+0x77:           popq   %rbx
io_watch_stdout+0x78:           popq   %r12
io_watch_stdout+0x7a:           popq   %r13
io_watch_stdout+0x7c:           popq   %rbp
> print regs
mdb: failed to dereference symbol: unknown symbol name
> ::regs
%rax = 0x0000000000000000       %r8  = 0x0000000000000000
%rbx = 0x0000000000963eb0       %r9  = 0x0000000000000000
%rcx = 0xfffffd7fffdfdcd0       %r10 = 0x0000000000000000
%rdx = 0x000000000040ec07       %r11 = 0x0000000000000246
%rsi = 0x0000000000000010       %r12 = 0x000000000067e690
%rdi = 0x000000000040ea4e       %r13 = 0x00000000005111b0
                                %r14 = 0x00000000009deeb0
                                %r15 = 0xfffffd7feecd7440

%cs = 0x0053    %fs = 0x0000    %gs = 0x0000
%ds = 0x0000    %es = 0x0000    %ss = 0x004b

%rip = 0x00000000004095ea io_watch_stdout+0x5a
%rbp = 0xfffffd7fffdfdd50
%rsp = 0xfffffd7fffdfdcd0

%rflags = 0x00010202
  id=0 vip=0 vif=0 ac=0 vm=0 rf=1 nt=0 iopl=0x0
  status=<of,df,IF,tf,sf,zf,af,pf,cf>

%gsbase = 0x0000000000000000
%fsbase = 0xfffffd7feefc2a40
%trapno = 0xe
   %err = 0x4
> 
```

**Comment by lukefromdc** [comment]
I am not screwing with files like that on my main, bare metal system. I do not have any VM's and am working on a live system. Recovery is from a backup partition if it dies, changing passwords is as far as I will go

**Comment by pyhalov** [comment]
Perhaps, it's even not compulsory.

**Comment by lukefromdc** [comment]
Anyway, with or without this PR and with the last commit to https://github.com/mate-desktop/mate-control-center/pull/383 applied, I don't get any errors running the resulting build of mate-about-me from terminal. Builds and runs fine, but I cannot test it beyond that.

**Comment by pyhalov** [comment]
OK. Thanks.

**Comment by lukefromdc** [comment]
Remember Debian and OpenIndiana could be very different, too

File: capplets/about-me/mate-about-me-password.c [hunk 1]
```diff
@@ -436,8 +436,10 @@ io_watch_stdout (GIOChannel *source, GIOCondition condition, PasswordDialog *pdi
...
 
 	if (g_io_channel_read_chars (source, buf, BUFSIZE, &bytes_read, &error) != G_IO_STATUS_NORMAL) {
-		g_warning ("IO Channel read error: %s", error->message);
-		g_error_free (error);
+		if (error) {
+			g_warning ("IO Channel read error: %s", error->message);
+			g_error_free (error);
+		}
 
 		return TRUE;
```
Comment (sc0w) [inline capplets/about-me/mate-about-me-password.c]: `[return TRUE;]` doesn't need to be added into the condition `[if (error)]` ?

Comment (muktupavels) [reply capplets/about-me/mate-about-me-password.c]: Why?

**Pull request merged**